sympy/sets/sets.py

```diff
@@ -153,6 +153,181 @@ def isdisjoint(self, other):
         """
         return self.is_disjoint(other)
 
+    def _union_simplify(self, other):
```

Comment:
I saw union of imageset is coming here [_union](https://github.com/sympy/sympy/pull/11188/files#diff-7b3e9b0852f1b1d094f87233f56e201fR331). 
Should we have `_union` in `fancysets/ImageSet` like we have [`_intersection`](https://github.com/Shekharrajak/sympy/blob/fae9f28623fb9bd81baef08cc0582aba9034a26f/sympy/sets/fancysets.py#L366) ?